### PR TITLE
Enabled support for nested block quotes

### LINKED ISSUE #9210: Add support for nested blockquotes
As CKEditor 4 allows creating nested blockquotes, due to the content compatibility we should make CKEditor 5 retain nested blockquote. Currently intentionally disabled in the schema. Makes content simpler (RTC, versioning, TC).

*   Does it need configuration (opt-in or opt-out)?
*   Where will the configuration be placed? Note: there's an identical scenario in tables – perhaps it should be somehow centralized? But where? 😕

---

If you'd like to see this feature implemented, add a 👍 reaction to this post.

### PR DESCRIPTION
**Pull request by psmyrek**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/contributing/git-commit-message-convention.html))

Feature (block-quote): Enabled support for nested block quotes. Closes #9210.

---

### Additional information

Support for nested block quotes in CF has a dedicated follow-up: https://github.com/cksource/collaboration-features/issues/3892.
After a brief check, I wasn't able to blow up the editor in CF samples.

### CODE REVIEW

**Comment by psmyrek** [comment]
PR is ready for testing @ckeditor/qa-team.

**Comment by FilipTokarski** [comment]
Looks fine, I didn't find any bugs 👍

**Comment by Reinmar** [comment]
Could you update docs and manual tests in a similar way I did in #9362?

**Comment by psmyrek** [comment]
Sure. I've just added a short info in migration guide, a new section in block quote docs how to revert the old behavior and a manual test with the old behavior.

File: docs/builds/guides/migration/migration-to-28.0.0.md [hunk 1]
```diff
@@ -14,4 +14,10 @@ Listed below are the most important changes that require your attention when upg
 
 Prior to version 28.0.0 inserting a table into another table was not allowed.
 
-If you wish to bring back this restriction see {@link features/table#disallowing-nesting-tables "Disallowing nesting tables"}.
\ No newline at end of file
+If you wish to bring back this restriction see {@link features/table#disallowing-nesting-tables "Disallowing nesting tables"}.
+
+## Disallowing nesting block quotes
+
+CKEditor v28.0.0 supports a block quote inserted into another block quote.
```
Comment (Reinmar) [inline docs/builds/guides/migration/migration-to-28.0.0.md:21]:
```suggestion
Prior to version 28.0.0 inserting block quote into a block quote was not allowed.
```

**Pull request merged**